app/components/shared/inputs/TimerInput.vue.ts

```diff
@@ -0,0 +1,132 @@
+import { Component, Prop } from 'vue-property-decorator';
+import { BaseInput } from './BaseInput';
+import { ITimerMetadata } from './index';
+
+let holdTimeout: number = null;
+let holdInterval: number = null;
+
+@Component({})
+export default class TimerInput extends BaseInput<number, ITimerMetadata> {
+
+  @Prop() value: number;
+  @Prop() metadata: ITimerMetadata;
+
+  hour = 3600;
+  minute = 60;
+  second = 1;
+  showTimerDropdown = false;
+  max = this.metadata.max || 3600;
+  min = this.metadata.min || 0;
+  format = this.metadata.format || 'ms';
+  hasHours = /h/.test(this.format);
+  hasSeconds = /s/.test(this.format);
+
+  get hours() {
+    return this.generateTime(this.max / 3600);
+  }
+
+  get minutes() {
+    return this.generateTime(60);
+  }
+
+  get seconds() {
+    return this.generateTime(60);
+ }
+
+  hideTimerDropdown(){
+    this.showTimerDropdown = false;
+  }
+
+  generateTime(time: number){
+    const times: string[] = [];
+
+    for(let num = 0; num < time; num++) {
+      const fill = num > 9 ? '' : '0';
+      times.push(fill + num);
+    }
+
+    return times;
+  }
+
+  isActiveHour(hour: number) {
+    const currentHour = Math.floor(this.value / 3600);
+    return (hour == currentHour);
+  }
+
+  isActiveMinute(minute: number) {
+    const currentMinute = Math.floor((this.value % 3600) / 60);
+    return (minute == currentMinute);
+  }
+
+  isActiveSecond(second: number) {
+    const currentSecond = Math.floor(((this.value % 3600) % 60) /60);
+    return second == currentSecond;
+  }
+
+  setHour(val: number) {
+    const currentMinsInSecs = Math.floor(this.value % 3600);
+    const hour = val * 3600;
+    this.updateValue(currentMinsInSecs + hour);
+  }
+
+  setMinute(val: number) {
+    const currentHrsInSecs = Math.floor(this.value / 3600) * 3600;
+    const minute = val * 60;
+    this.updateValue(currentHrsInSecs + minute);
+  }
+
+  setSecond(val: number) {
+    const currentMinsInSecs = Math.floor((this.value % 3600) / 60) * 60;
+    this.updateValue(currentMinsInSecs + (val * 1));
+  }
+
+  updateValue(value: number) {
```

Comment:
use `emitInput` from `BaseInput`. I think you took this method somewhere from `SliderInput`, I'm going to fix that in SliderInput too.

Reply:
again, i think this was copied over from the site code